src/utils/job-manager.js

```diff
@@ -68,14 +71,22 @@ export default class JobSocket extends EventTarget {
         this._socket.send(JSON.stringify(msg));
     }
 
-    _onError(event) {
+    _onOpenOrClose = (event) => {
```

Comment:
I'd break this function into `_onOpen` and `_onClose` and register them separately.